daemon/images/image_commit.go

```diff
@@ -12,7 +13,7 @@ import (
 )
 
 // CommitImage creates a new image from a commit config
-func (i *ImageService) CommitImage(c backend.CommitConfig) (image.ID, error) {
+func (i *ImageService) CommitImage(ctx context.Context, c backend.CommitConfig) (image.ID, error) {
```

Comment:
This function does not actually handle the context (adding it here is mostly to satisfy the interface). ISTR at some point we discussed "that's okay, we never did, so let's implement it only for the new (containerd integration) code, and only for parts that are relevant for that). (/cc @corhere to keep me honest).

If that's still the case, perhaps make that explicit;

```suggestion
func (i *ImageService) CommitImage(_ context.Context, c backend.CommitConfig) (image.ID, error) {
```

And (maybe?) add a comment that describes that (if it makes sense to do so), so that we outline the expectations of passing a context here.

Reply:
Since context cancelation is inherently racy, the semantics cannot be assumed to be anything stronger than a hint that the result of the operation is no longer required. Furthermore, accepting a context cannot on its own imply that the function call respects cancelation: the function might only accept a context for the request-scoped values. From the caller's point of view, unless explicitly documented otherwise, a function which completely ignores its context argument _is_ handling the context! Ignoring the argument is an implementation detail which is subject to change, and should therefore be kept out of the `CommitImage` API. I think the argument should be named `ctx` as callers should still be prepared for the function call to be canceled when passed a cancelable context.

Also, there is a trivial way to make this function handle the context:
```suggestion
func (i *ImageService) CommitImage(ctx context.Context, c backend.CommitConfig) (image.ID, error) {
	if err := ctx.Err(); err != nil {
		return "", err
	}
```

Reply:
>  Furthermore, accepting a context cannot on its own imply that the function call respects cancelation: the function might only accept a context for the request-scoped values. From the caller's point of view, unless explicitly documented otherwise, a function which completely ignores its context argument is handling the context! 

Right, so perhaps a comment wouldn't be needed; I _do_ think it can help transparency (and help with some linters) to use `_` for unused arguments.

I like your suggestion though, and perhaps something to consider adding; wdyt @vvoland?

Reply:
That sounds good to me. I will add this one, thanks!

Reply:
Done